notify/impl.go

```diff
@@ -680,12 +685,27 @@ func (n *OpsGenie) Notify(ctx context.Context, as ...*types.Alert) error {
 	if err != nil {
 		return err
 	}
-	resp.Body.Close()
+	defer resp.Body.Close()
 
-	if resp.StatusCode/100 != 2 {
+	if resp.StatusCode == 400 && alerts.Status() == model.AlertResolved {
 		body, _ := ioutil.ReadAll(resp.Body)
-		log.With("incident", key).Debugf("unexpected OpsGenie response %s: %s", resp.Status, body)
-		return fmt.Errorf("unexpected status code %v", resp.StatusCode)
+
+		var responseMessage opsGenieErrorResponse
+		if err := json.Unmarshal(body, &responseMessage); err != nil {
+			return fmt.Errorf("OpsGenie: Could not parse error response \"%s\"", body)
+		}
+		// Response code 5 appears to indicate that an alert was already closed when we tried to close it.
+		// Consider that a success.
+		if responseMessage.Code == 5 {
```

Comment:
Can we define a constant for that magic number? Probably removes the need for a comment as well.